source/manual/architecture-shallow-dive.html.md

```diff
@@ -0,0 +1,204 @@
+
+Within Origin, the request goes to an application called Router to determine which GOV.UK application will render the page.
+
+Router gets the request and decides which application should render the page. Router then proxies (forwards) the request to the Government Frontend application.
+
+Government Frontend looks at request and decides which content to get from the Content Store. Government Frontend then queries the Content Store API to get the content in an [API response from the Content Store](https://www.gov.uk/api/content/government/news/uk-changes-travel-rules-for-travellers-from-nigeria-to-england).
+
+Government Frontend then builds the page using the content in the API response, and sends that response back to Router.
+
+Router sends the response to Fastly. Fastly sends the response back to the user.
+
+### A user searches GOV.UK using the search page
+
+The user goes to the GOV.UK search page, and searches for something.
+
+The user's web browser checks if it has a locally cached version of the search results by sending a request to the Fastly content delivery network (CDN). A locally cached version is more likely if the user has made the same search in the last 5 mins.
+
+If Fastly has the search results page cached, Fastly returns the results page to the user.
+
+If Fastly does not have the search results page cached, Fastly sends the request on to Origin. Origin is the infrastructure that GOV.UK maintains, hosted in Amazon Web Services (AWS).
+
+Within Origin, the request goes to an application called Router. The Finder Frontend application serves search results pages, so Router sends the request to Finder Frontend.
+
+Finder Frontend queries the Search API. Search API provides a [JSON response with the search results](https://www.gov.uk/api/search.json?q=covid&count=2). Finder Frontend then renders the response in a search results page to the user.
+
+## A publisher publishes content on GOV.UK
+
+GOV.UK has the following applications to publish content:
+
+- Mainstream Publisher
+- Whitehall
+- Content Publisher
+- Specialist Publisher
+- Travel Advice Publisher
+- Collections Publisher
+
+The following content states what happens when a user publishes content to live and to draft using Mainstream Publisher ("Publisher").
+
+### A user publishes content to live
+
+A user signs into their Signon account. The user creates or changes content in Publisher and saves that content in draft.
+
+Publisher makes a request to Publishing API to create or update the appropriate draft content.
+
+When the user is ready to publish the content, they press __Publish__ in the Publisher which sends a request to Publishing API.
+
+The Publishing API then updates the Content Store with the new or updated content.
+
+Content Store updates Router to say there is now content at a particular URL and says which frontend application will render that content.
+
+The Publishing API tells the Search API that the content is live.
```

Comment:
The 'message queue' is quite important, so I've filled out that section a bit:

```suggestion
Publishing API updates the Content Store directly, which in turn updates Router to say there is now content at a particular URL, and which frontend application will render that content.

Publishing API also puts the publishing event on a 'message queue', which a number of other applications rely on. For example, Search API listens to this queue, updating its indexes when new content is published, so that it can be searched for by users. The Email Alert Service also listens to this queue, so that it can send emails to users who have subscribed to that topic.
```